files/image_config/hostcfgd/hostcfgd

```diff
@@ -41,12 +41,13 @@ def obfuscate(data):
         return data
 
 
-def update_feature_state(feature_name, state):
+def update_feature_state(feature_name, state, has_timer=False):
+    feature_suffix = "timer" if has_timer else "service"
```

Comment:
@tahmed-dev wondering is it it not better to check for timer service exits for any service and start it if present? That way it can be dynamic and we don't to pre-define in init_cfg.json.j2 as this is always can break if new service is added but init_cfg file is not updated accordingly.

Reply:
@abdosi that would work as well. The argument is applicable to features as well. This would also break a feature as it will not be started. I think it is simpler for hostcfgd to not assume any knowledge about systemd internals or where .service/.timer files are on disk had in chance systemd relocated those files. After all this is one time configuration and it should be well defined during development.

After all, if you feel strongly about it, please go ahead and put out a PR to that effect.

Reply:
@abdoi: I made the same suggestion [above[(https://github.com/Azure/sonic-buildimage/pull/5211#discussion_r473262514). I think the current solution (relying on init_cfg.json) is better than explicitly specifying the names of the services which have a .timer file. I'm still open to checking for the presence of a .timer file.

Reply:
Thanks @tahmed-dev and @jleveque. 
I was thinking just  check return value of below command and based on that use either .service or .timer 
"sudo "sudo systemctl list-unit-files | grep {}.timer".format(feature_name)

Reply:
@abdosi Thanks! I did not know about this command.

The only thing that would hold me off is that this comes with a cost during boot time as such check for every service will consume precious CPU cycles in this path (boot time).

Reply:
@tahmed-dev: I was also concerned about that downside to checking for the unit files. I guess we could check the runtime of the `sudo systemctl list-unit-files | grep ...` command to understand how intensive it is. But as above, I'm OK with expecting this information to be added to init_cfg.json -- now, all new services should be added there. It's one location and it's a data file. What I really wanted to avoid (and this implementation does that) is the need to add new service names into various code files if they are exceptions to the norm (e.g., they have a .timer file).

Reply:
Had offline chat with @tahmed-dev and since using this approach can have boot-time impact so we park this discussion or now .

Reply:
OK. Definitely something to reconsider in the future.